preprocess_observation.py

```diff
@@ -21,25 +21,69 @@ def preprocess_observation_helper(mydict):
     for pokemon in mydict['side']['pokemon']:
         name = pokemon['ident'][4:]
         tempDict = {'Aggron': 0, 'Arceus': 1, 'Cacturne': 2, 'Dragonite': 3, 'Druddigon': 4, 'Uxie': 5}
-        health = int(pokemon['condition'].split('/')[0].split(' ')[0])
-        retval.append([809*2 + tempDict[name], health])
-
-        index = pokedex[name.lower()]['num']
-        if pokemon['active']:
-            retval.append([index-1, 1])
+        # Add health information for this pokemon
+        condition = pokemon['condition'].split('/')[0].split(' ')
+        health = int(condition[0])
+        retval.append([NUM_POKEMON*2 + tempDict[name], health])
+        # Add status information for this Pokemon
+        if len(condition) != 1 and condition[1] != 'fnt':
+            # in the future, a numerical value (e.g. 2 turns of sleep remaining) would be nice instead of just 1/0.
+            # TODO: confusion is not mutually exclusive with other status conditions, take account of this
+            assert(len(condition) == 2)
+            assert(condition[1] in STATUS_DICT)
+            for i in range(7):
```

Comment:
Why 7? Make this a named constant or variable.

Reply:
This is a reflection of the fact that NUM_STATUS_CONDITIONS and STATUS_DICT in game_model.py are not the same length. Put resolving this into todo list.